src/appshell/view/dockwindow/dockpageview.cpp

```diff
@@ -207,36 +221,48 @@ void DockPageView::setDockOpen(const QString& dockName, bool open)
     if (!open) {
         dock->close();
         return;
-    }
+    } else {
+        dock->setFocus(true);
 
-    DockPanelView* panel = dynamic_cast<DockPanelView*>(dock);
-    if (!panel) {
-        dock->open();
-        return;
-    }
+        DockPanelView* panel = dynamic_cast<DockPanelView*>(dock);
+        if (!panel) {
+            dock->open();
+            return;
+        }
 
-    DockPanelView* destinationPanel = findPanelForTab(panel);
-    if (destinationPanel) {
-        const int currentTabIndex = destinationPanel->currentTabIndex();
-        const int tabIndex = destinationPanel->tabIndexOfPanel(panel);
-        const bool tabNotFound = tabIndex < 0;
-        const bool tabNotCurrent = tabIndex != currentTabIndex;
-        if (tabNotFound) {
-            destinationPanel->addPanelAsTab(panel);
-        } else if (tabNotCurrent) {
-            destinationPanel->setCurrentTabIndex(tabIndex);
+        if (!panel->isOpen()) {
+            panel->open();
```

Comment:
As it turns out, if you simply open a closed panel, it will open at its last closed location, which is great.